R/drop_name_crossref.R

```diff
@@ -0,0 +1,46 @@
+#' Create visual citation from Crossref data
+#'
+#' Takes one or several dois and extracts information from Crossref, then processes them into visual citations with \code{\link{drop_name}}
+#'
+#' @param dois One or several dois to create visual citations for. If they are named, these names are used as filenames; otherwise they are generated based on authors and years.
+#' @inheritDotParams drop_name -bib -cite_key
+#'
+#' @export
+
+drop_name_crossref <- function(dois, ...) {
+
+  if (!"rcrossref" %in% rownames(installed.packages())) {
+    stop("This function requires the rcrossref package. Please install and retry.")
+  }
+
+  stopifnot(is.character(dois))
+
+  ref <-  rcrossref::cr_cn(dois, "citeproc-json")
```

Comment:
What would you say, if we wrapped this into a try-catch environment? If this fails, we could provide solutions to common issues, such as not providing one's mail address to cross-ref, etc..

Reply:
Yes - even though my understanding is that providing the email address is optional. But given the issues you experienced that seems necessary.

Reply:
I haven't used the rcrossref package before, so Im not too deep into the details. I came across [one stackoverflow answer](https://stackoverflow.com/questions/68412035/rcrossref-api-error-resresponse-headerscontent-type-application-jsonch) that made it seem it's mandatory nowadays. (The linked github issue doesn't say that, though). Better safe than sorry.